lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart

```diff
@@ -0,0 +1,110 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_state_support.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class ShowWfsLayerWithXmlQuery extends StatefulWidget {
+  const ShowWfsLayerWithXmlQuery({super.key});
+
+  @override
+  State<ShowWfsLayerWithXmlQuery> createState() =>
+      _ShowWfsLayerWithXmlQueryState();
+}
+
+class _ShowWfsLayerWithXmlQueryState extends State<ShowWfsLayerWithXmlQuery>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Stack(
+        children: [
+          Column(
+            children: [
+              Expanded(
+                // Add a map view to the widget tree and set a controller.
+                child: ArcGISMapView(
+                  controllerProvider: () => _mapViewController,
+                  onMapViewReady: onMapViewReady,
+                ),
+              ),
+            ],
+          ),
+          // Display a progress indicator and prevent interaction until state is ready.
+          Visibility(
+            visible: !_ready,
+            child: const SizedBox.expand(
+              child: ColoredBox(
+                color: Colors.white30,
+                child: Center(child: CircularProgressIndicator()),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISNavigation);
+    _mapViewController.arcGISMap = map;
+
+    // Load the WFS layer with the XML query.
+    await loadWfsLayerWithXmlQuery();
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> loadWfsLayerWithXmlQuery() async {
+    const wfsFeatureTableUri =
+        'https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&amp;request=getcapabilities';
+
+    // Create the WFS feature table from URI and name.
+    final statesTable = WfsFeatureTable.withUriAndTableName(
+      uri: Uri.parse(wfsFeatureTableUri),
+      tableName: 'Seattle_Downtown_Features:Trees',
+    )
+      // Set the feature request mode and axis order.
+      ..axisOrder = OgcAxisOrder.noSwap
+      ..featureRequestMode = FeatureRequestMode.manualCache;
+
+    // Create the feature layer from the feature table.
+    final featureLayer = FeatureLayer.withFeatureTable(statesTable);
+    await featureLayer.load();
```

Comment:
Do you need to actively load the layer, or would adding it to the map's operationalLayers be enough?

Reply:
we load here specifically to instantiate the layer to ensure the query and full extent correctly.